src/web/mjs/connectors/Copymanga.mjs

```diff
@@ -1,16 +1,117 @@
 import Connector from '../engine/Connector.mjs';
 import Manga from '../engine/Manga.mjs';
 
+function formatDate() {
+    // Example: 2023.03.27
+    const now = new Date();
+    const year = now.getFullYear();
+    const month = now.getMonth() + 1;
+    const day = now.getDate();
+    return `${year}.${month < 10 ? "0" : ""}${month}.${day}`;
+}
+
+const API_HEADERS = {
+    'User-Agent': '"User-Agent" to "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.5005.124 Safari/537.36 Edg/102.0.1245.44"',
+    'version': formatDate(),
+    "platform": "1",
+};
+
 export default class Copymanga extends Connector {
 
     constructor() {
         super();
         super.id = 'copymanga';
         super.label = '拷貝漫畫 (Copymanga)';
         this.tags = [ 'manga', 'webtoon', 'chinese' ];
-        this.url = 'https://copymanga.com';
-        this.apiurl = 'https://api.copymanga.com';
+        this.config = {
+            url: {
+                label: 'Website URL',
+                description: 'This website may change their URL.\nThis is the last known URL which can also be manually set by the user.',
+                input: 'text',
+                value: 'https://www.copymanga.site'
+            },
+            apiurl: {
+                label: 'API URL',
+                description: 'This website may change their API URL.\nThis is the last known API URL which can also be manually set by the user.',
+                input: 'text',
+                value: 'https://api.copymanga.org'
+            },
+            format:  {
+                label: 'Preferred format',
+                description: 'format of images\nwebp \njpg',
+                input: 'select',
+                options: [
+                    { value: 'webp', name: 'webp' },
+                    { value: 'jpg', name: 'jpg' },
+                ],
+                value: 'webp'
```

Comment:
The `jpeg` format should be default, almost all HakuNeko users are asking to get JPEG instead of WEBP